pkg/webreg/webreg_test.go

```diff
@@ -165,3 +165,150 @@ func TestGetDependencyDataItems(t *testing.T) {
+			Environment: []api.StepParameter{
+				{
+					Name:          "var2",
+					Documentation: "var2 documentation",
+				},
+			},
+		},
+	}
+
+	registrySteps := registry.ReferenceByName{
+		stepWithoutVars.As: *stepWithoutVars.LiteralTestStep,
+		step1.As:           *step1.LiteralTestStep,
+		step2.As:           *step2.LiteralTestStep,
+		step3.As:           *step3.LiteralTestStep,
+	}
+
+	stepPtr := func(name string) *api.LiteralTestStep {
+		if step, ok := registrySteps[name]; !ok {
+			t.Fatalf("TEST BUG: step '%s' not in test registry", name)
+			return nil
+		} else {
+			return &step
+		}
+	}
+
+	registryChains := registry.ChainByName{
+		"chain-with-step-1": api.RegistryChain{
+			As:    "chain-with-step-1",
+			Steps: []api.TestStep{{LiteralTestStep: stepPtr("step-1")}},
+		},
+		"chain-with-step1-and-step-2": api.RegistryChain{
+			As:    "chain-with-step-1",
+			Steps: []api.TestStep{{LiteralTestStep: stepPtr("step-1")}, {LiteralTestStep: stepPtr("step-2")}},
+		},
+		"chain-with-step1-and-step-2-and-step-3": api.RegistryChain{
+			As:    "chain-with-step-1",
+			Steps: []api.TestStep{{LiteralTestStep: stepPtr("step-1")}, {LiteralTestStep: stepPtr("step-2")}, {LiteralTestStep: stepPtr("step-3")}},
+		},
+	}
+
+	testCases := []struct {
+		description string
+		inputSteps  []api.TestStep
+
+		expected map[string]environmentLine
+	}{
+		{
+			description: "no input, no data",
+			expected:    map[string]environmentLine{},
+		},
+		{
+			description: "Chain with single step without vars, no data",
+			inputSteps:  []api.TestStep{stepWithoutVars},
+			expected:    map[string]environmentLine{},
+		},
+		{
+			description: "Chain with single step with value",
+			inputSteps:  []api.TestStep{{LiteralTestStep: stepPtr("step-1")}},
+			expected: map[string]environmentLine{
+				"var1": {
+					Documentation: "var1 documentation",
+					Default:       &defaultVal,
+					Steps:         []string{"step-1"},
+				},
+			},
+		},
+		{
+			description: "Chain with two steps with value",
+			inputSteps:  []api.TestStep{{LiteralTestStep: stepPtr("step-1")}, {LiteralTestStep: stepPtr("step-2")}},
+			expected: map[string]environmentLine{
+				"var1": {
+					Documentation: "var1 documentation",
+					Default:       &defaultVal,
+					Steps:         []string{"step-1"},
+				},
+				"var2": {
+					Documentation: "var2 documentation",
+					Steps:         []string{"step-2"},
+				},
+			},
+		},
+		{
+			description: "Chain with three steps with two steps having same value",
+			inputSteps:  []api.TestStep{{LiteralTestStep: stepPtr("step-1")}, {LiteralTestStep: stepPtr("step-2")}, {LiteralTestStep: stepPtr("step-3")}},
+			expected: map[string]environmentLine{
+				"var1": {
+					Documentation: "var1 documentation",
+					Default:       &defaultVal,
+					Steps:         []string{"step-1"},
+				},
+				"var2": {
+					Documentation: "var2 documentation",
+					Steps:         []string{"step-2", "step-3"},
+				},
+			},
+		},
+	}
+
+	t.Parallel()
```

Comment:
Go testing nti: this line says "run this test (and all of it's sub-tests) in parallel with _other tests_. It does _not_ say "run these sub-tests in parallel". If you also want that, you need to the call the `Parallel()` method on the `t` you get in the closure passed to `t.Run()`